src/Location.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from "react";
+
+export default class Location extends Component {
+	constructor() {
+		super();
+		this.state = {
+			allLocations: [],
+			display: true,
+		};
+	}
+	fetchLocation = () => {
+		fetch("https://ghibliapi.herokuapp.com/locations")
+			.then((res) => res.json())
+			.then((data) => {
+				this.setState({
+					allLocations: data,
+					display: true,
+				});
+			});
+	};
+	hideLocation = () => {
+		this.setState({
+			display: false,
+		});
+	};
+	componentDidMount = () => {
+		this.fetchLocation();
+	};
+	render() {
+		let allLoco = this.state.allLocations.map((each) => {
+			return (
+				<div>
+					<ul>
+						<li>Name:{each.name}</li>
+					</ul>
+					<div>Climate{each.climate}</div>
+					<div>Terrain{each.terrain}</div>
+				</div>
+			);
+		});
+
+		return (
+			<div>
+				<h1>List of Locations</h1>
+				{this.state.display && (
+					<button onClick={this.hideLocation}>Show Locations</button>
+				)}
+				{!this.state.display && (
+					<button onClick={this.fetchLocation}>Hide Locations</button>
+				)}
+				<div className="locations">{this.state.display ? null : allLoco}</div>
+			</div>
+		);
```

Comment:
```suggestion
		return (
			<div>
				<h1>List of Locations</h1>
				
					<button onClick={this.toggleLocation}>
					{this.state.display ? "Hide Locations" : "Show Locations"} //<- you can change the button's text content based on whether this.state.display is true or false.
					</button>
					
				<ul className="locations">{this.state.display ? allLoco : null}</ul>
			</div>
		);
```
1) The "Show/Hide" button can be a single button that uses a single event handler to toggle `this.state.display` back and forth between **true** and **false**. You can change the button's text content based on whether `this.state.display` is **true** or **false**.

2) If `this.state.display` is **true**, you should display _the list of locations_, not display _**null**_. The reason you had to write your conditional logic backwards like this is because of the way you set state in `this.fetchLocation` and `this.hideLocation`.